hw/top_earlgrey/data/clocks.xdc

```diff
@@ -0,0 +1,23 @@
+## Copyright lowRISC contributors.
+## Licensed under the Apache License, Version 2.0, see LICENSE for details.
+## SPDX-License-Identifier: Apache-2.0
+
+## Clock Signal
+create_clock -add -name sys_clk_pin -period 10.00 -waveform {0 5} [get_ports IO_CLK]
+
+## Clock Domain Crossings
+create_generated_clock -name clk_50_unbuf -source [get_pin clkgen/pll/CLKIN1] [get_pin clkgen/pll/CLKOUT0]
+create_generated_clock -name clk_48_unbuf -source [get_pin clkgen/pll/CLKIN1] [get_pin clkgen/pll/CLKOUT1]
+set_clock_groups -group clk_50_unbuf -group clk_48_unbuf -asynchronous
+
+## Divided clock
+## This is not really recommneded per Vivado's guidelines, but hopefully these clocks are slow enough and their
+## destination flops few enough.
+
+create_generated_clock -name clk_io_div2 -source [get_pin clkgen/pll/CLKOUT0] -divide_by 2 \
+   [get_pin top_earlgrey/u_clkmgr/u_io_div2_div/gen_div2.u_div2/gen_generic.u_impl_generic/q_o_reg[0]/Q]
+
+create_generated_clock -name clk_io_div4 -source [get_pin clkgen/pll/CLKOUT0] -divide_by 4 \
```

Comment:
this is probably ok since we define the source and derived clocks to be asynchronous further below.
otherwise we could also use the CK pin of the divider FF as source pin in order to improve the accuracy of the derived phase
`get_pin top_earlgrey/u_clkmgr/u_io_div4_div/gen_div.clk_int_reg/C`

Reply:
updated!